components/collector/src/source_collectors/gitlab/source_up_to_dateness.py

```diff
@@ -68,3 +72,35 @@ async def _parse_value(self, responses: SourceResponses) -> Value:
         commit_responses = responses[1:]
         commit_dates = [parse((await response.json())["committed_date"]) for response in commit_responses]
         return str(days_ago(max(commit_dates)))
+
+
+class GitLabPipelineUpToDateness(TimePassedCollector, GitLabJobsBase):
+    """Collector class to measure the up-to-dateness of a job/pipeline."""
+
+    async def _landing_url(self, responses: SourceResponses) -> URL:
+        """Override to return a landing URL for the pipeline."""
+        if responses and (json := await responses[0].json()):
+            return URL(json[0]["pipeline"]["web_url"])
+        return await super()._landing_url(responses)
+
+    async def _parse_source_response_date_time(self, response: Response) -> datetime:
```

Comment:
Is it not easier to return a `datetime.date` object, as we only report on amount of days anyway?

Reply:
I'd have to check whether this is also used in situations where we do want time.